python/ray/serve/config.py

```diff
@@ -17,86 +19,59 @@ def _callable_is_blocking(func_or_class):
         return not inspect.iscoroutinefunction(func_or_class.__call__)
 
 
-class BackendConfig:
-    def __init__(self, config_dict, accepts_batches=False, is_blocking=True):
-        assert isinstance(config_dict, dict)
-        # Make a copy so that we don't modify the input dict.
-        config_dict = config_dict.copy()
-
-        self.accepts_batches = accepts_batches
-        self.is_blocking = is_blocking
-        self.num_replicas = config_dict.pop("num_replicas", 1)
-        self.max_batch_size = config_dict.pop("max_batch_size", None)
-        self.batch_wait_timeout = config_dict.pop("batch_wait_timeout", 0)
-        self.max_concurrent_queries = config_dict.pop("max_concurrent_queries",
-                                                      None)
-        self.autoscaling_config = config_dict.pop("autoscaling", None)
-
-        if self.max_concurrent_queries is None:
+class BackendConfig(BaseModel):
+    accepts_batches: bool = False
+    is_blocking: bool = True
+    num_replicas: PositiveInt = 1
+    max_batch_size: Optional[PositiveInt] = None
+    batch_wait_timeout: float = 0
+    max_concurrent_queries: Optional[int] = None
+    autoscaling_config: Optional[Dict[str, Any]] = None
```

Comment:
Can we avoid exposing this to users for now somehow? Else we should mark it as experimental somehow.

Also, is it possible to add a description for each of these fields that will be auto-documented?

Reply:
You're just talking about the `autoscaling_config`?  Would putting an underscore before the name be enough?

For the auto-documentation, yes, I'll add it.  And the autodoc will also hide variables whose name starts with an underscore.

Reply:
underscore sounds good.

Reply:
It turns out that with our current version of Spnihx (1.8.5), there's no way to autodocument type hints, nor is there a way to autodocument default values for instance attributes.  So I'll just use a docstring for the class `BackendConfig`.